File: Problems/No.225_Implement_Stack_using_Queues/implement_stack_using_queues.cpp
class MyStack {

private:
    std::queue<int> Queue;

public:
    
    void push(int x) { Queue.push(x); }
    
    int pop() {

        int item;

        for (int i=0; i<Queue.size()-1; i++) {

            Queue.push(Queue.front());
            Queue.pop();

        }

        item = Queue.front();
        Queue.pop();

        return item;
        
    }
    
    int top() { return Queue.back(); }
    
    bool empty() { return Queue.empty(); }

};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: Problems/No.225_Implement_Stack_using_Queues/implement_stack_using_queues.py
class MyStack(object):

    def __init__(self):
        self.Stack = []

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.Stack.append(x)

    def pop(self):
        """
        :rtype: int
        """
        item = self.Stack.pop()
        return item

    def top(self):
        """
        :rtype: int
        """
        return self.Stack[-1]

    def empty(self):
        """
        :rtype: bool
        """
        return True if not self.Stack else False


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()
